packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -35,134 +39,143 @@ interface InMemoryAnchorConfig {
-          await this.verifySignedCommit(record)
-        }
-
-        const log = await this._loadCommitHistory(req.cid)
-        const candidate = new Candidate(new CID(req.cid), req.docId, log)
+          const log = await this._loadCommitHistory(req.cid);
+          const candidate = new Candidate(req.cid, req.docId, log);
 
-        if (!result[candidate.key]) {
-          result[candidate.key] = []
+          if (!result[candidate.key]) {
+            result[candidate.key] = [];
+          }
+          result[candidate.key].push(candidate);
+        } catch (e) {
+          console.error(e.message);
+          this._failCandidate(req, e.message);
         }
-        result[candidate.key].push(candidate)
-      } catch (e) {
-        console.error(e.message)
-        await this._failCandidate(req, e.message)
-      }
-    }
-
-    return result
+      })
+    );
+    return result;
   }
 
-  async _selectValidCandidates(groupedCandidates: Record<string, Candidate[]>): Promise<Candidate[]> {
-    const result: Candidate[] = []
+  _selectValidCandidates(
+    groupedCandidates: Record<string, Candidate[]>
+  ): Candidate[] {
+    const result: Candidate[] = [];
     for (const compositeKey of Object.keys(groupedCandidates)) {
-      const candidates: Candidate[] = groupedCandidates[compositeKey]
+      const candidates = groupedCandidates[compositeKey];
 
       // When there are multiple valid candidate tips to anchor for the same docId, pick the one
       // with the largest log
-      let selected: Candidate = null
+      let selected: Candidate = null;
       for (const c of candidates) {
         if (selected == null) {
-          selected = c
-          continue
+          selected = c;
+          continue;
         }
 
         if (c.log.length < selected.log.length) {
-          await this._failCandidate(c)
+          this._failCandidate(c);
         } else if (c.log.length > selected.log.length) {
-          await this._failCandidate(selected)
-          selected = c
+          this._failCandidate(selected);
+          selected = c;
         } else {
           // If there are two conflicting candidates with the same log length, we must choose
           // which to anchor deterministically. We use the same arbitrary but deterministic strategy
           // that js-ceramic conflict resolution does: choosing the record whose CID is smaller
           if (c.cid.bytes < selected.cid.bytes) {
-            await this._failCandidate(selected)
-            selected = c
+            this._failCandidate(selected);
+            selected = c;
           } else {
-            await this._failCandidate(c)
+            this._failCandidate(c);
           }
         }
       }
 
-      result.push(selected)
+      result.push(selected);
     }
 
-    return result
+    return result;
   }
 
-  async _failCandidate(candidate: Candidate, message?: string): Promise<void> {
+  _failCandidate(candidate: Candidate, message?: string): void {
```

Comment:
So I had read somewhere when I was first teaching myself about javascript/typescript a few months ago that it's a bad idea to mix sync and async calls.  I don't totally remember why it was a problem though, something about how the node event loops works.  Do you know what I'm referring to?  Is this something I shouldn't actually be worried about?

Reply:
Why is that relevant here? This function is completely sync so it should not be async.

Reply:
https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/ - irrelevant here. Nothing here relies on event-loop.

Reply:
Yeah, that must have been what I was remembering.  Guess I was overthinking it.  Thanks!